### PR TITLE
New WebGLNodeBuilder + Nodes: transmission

### PR DESCRIPTION
**Pull request by sunag**
Related issue: https://github.com/mrdoob/three.js/pull/24385

**Description**

Part 4 / 4

I rewrote the Slots API and I added `transmission*Node` and `specular*Node`, `iorNode`. 
Now `WebGLNodeBuilder` can detect the precedence of the slots and added the code/snippet always in the correct sequence.

*This contribution is funded by [Google via Igalia](https://igalia.com)*

### CODE REVIEW

**Comment by mrdoob** [comment]
> 

Is that output correct?

**Comment by sunag** [comment]
> Is that output correct?

Yees! hehe :grin: I put a `normalWorld` in `colorNode`... do you think I should remove it?

**Comment by mrdoob** [comment]
> Yees! hehe :grin: I put a `normalWorld` in `colorNode`... do you think I should remove it?

Yeah, it looks like it's broken otherwise 😅

**Comment by sunag** [comment]
> Yeah, it looks like it's broken otherwise 😅

Ok, done!!

File: examples/webgl_nodes_loader_gltf_transmission.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+								const glassMesh = gltf.scene.getObjectByName( 'glassCover' );
+
+								const material = glassMesh.material;
+
+								if ( material && material.transmission > 0 ) {
+
+									const checkerNode = checker( mul( add( uv(), vec2( 0, timerLocal( - .05 ) ) ), 7 ) );
+
+									const nodeMaterial = NodeMaterial.fromMaterial( material );
+									nodeMaterial.transmissionNode = float( 1 );
```
Comment (LeviPesin) [inline examples/webgl_nodes_loader_gltf_transmission.html:88]: I think we can write just `1` and `1.5` without any `float()`.

Comment (sunag) [reply examples/webgl_nodes_loader_gltf_transmission.html:88]: I think that no! This can't support primitive values here, just Node classes.

**Comment by mrdoob** [comment]
> image

Still looks broken, in a different way now 😇

**Comment by sunag** [comment]
> Still looks broken, in a different way now 😇

Updated :)

**Comment by sunag** [comment]
Updated :)

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**